app/uk/gov/hmrc/tai/forms/IncomeCalculatorForm.scala

```diff
@@ -66,37 +66,41 @@ object HoursWorkedForm extends EditIncomeIrregularPayConstants {
 }
 
 case class PayPeriodForm(payPeriod: Option[String],
-                         otherInDays: Option[Int] = None)
+                         otherInDays: Option[String] = None) {
+}
 
-object PayPeriodForm{
+object PayPeriodForm extends EditIncomePayPeriodConstants{
   implicit val formats = Json.format[PayPeriodForm]
 
   def createForm(howOftenError: Option[String], payPeriod : Option[String] = None)(implicit messages: Messages): Form[PayPeriodForm] = {
 
     val payPeriodValidation = Constraint[Option[String]]("Please select a period"){
-
-      case Some(txt) => txt match {case "other" | "monthly" | "weekly" | "fortnightly" => Valid
-      case _ => Invalid(messages("tai.payPeriod.error.form.incomes.radioButton.mandatory"))
+      case Some(txt) => txt match {
+        case OTHER | MONTHLY | WEEKLY | FORTNIGHTLY => Valid
+        case _ => Invalid(messages("tai.payPeriod.error.form.incomes.radioButton.mandatory"))
       }
       case _ => Invalid(messages("tai.payPeriod.error.form.incomes.radioButton.mandatory"))
     }
 
-    def otherInDaysValidation(payPeriod : Option[String]) : Constraint[Option[Int]] = {
-      Constraint[Option[Int]]("days") {
-        days => {
-          if(payPeriod.getOrElse("") == "other" && !days.isDefined) {
-            Invalid(messages("tai.payPeriod.error.form.incomes.other.mandatory"))
-          } else {
-            Valid
+    def otherInDaysValidation(payPeriod : Option[String]) : Constraint[Option[String]] = {
+      val digitsOnly = """^\d*$""".r
+
+      Constraint[Option[String]]("days") {
+        days: Option[String] => {
+          (payPeriod, days) match {
+            case (Some(OTHER), Some(digitsOnly())) => Valid
+            case (Some(OTHER), None) => Invalid(messages("tai.payPeriod.error.form.incomes.other.mandatory"))
+            case (Some(OTHER), _) => Invalid(messages("tai.payPeriod.error.form.incomes.other.invalid"))
```

Comment:
It might be more expressive to have an explicit match on this, ie a `Some` with at least one invalid character